### PR TITLE
Allow AhcEndpoint address override in grizzly provider

### PR DESCRIPTION
**Pull request by elrodro83**
Add usage of Request#getInetAddress() in grizzly provider.

This allows for overriding the address where the physical connection is established, by setting it in the request instead of letting the provider use the host from the url or the proxy.

Netty is already doing this (https://github.com/AsyncHttpClient/async-http-client/blob/1.9.x/src/main/java/com/ning/http/client/providers/netty/request/NettyRequestSender.java#L354) among with other name resolution stuff, and it is a feature i am needing in grizzly provider.

### CODE REVIEW

File: src/main/java/com/ning/http/client/providers/grizzly/ConnectionManager.java [hunk 1]
```diff
@@ -219,10 +219,11 @@ static boolean isKeepAlive(final Connection connection) {
         return !IS_NOT_KEEP_ALIVE.isSet(connection);
     }
     
-    private static String getPartitionId(Request request,
+    private static String getPartitionId(InetAddress overrideAddress, Request request,
             ProxyServer proxyServer) {
-        return request.getConnectionPoolPartitioning()
-                .getPartitionKey(request.getUri(), proxyServer).toString();
+        return overrideAddress != null ? overrideAddress.toString() + "_" : "" +
```
Comment (oleksiys) [inline src/main/java/com/ning/http/client/providers/grizzly/ConnectionManager.java]: is it supposed to be
(overrideAddress != null ? overrideAddress.toString() + "_" : "") +
           request.getConnectionPoolPartitioning()
           .getPartitionKey(request.getUri(), proxyServer).toString();

?

**Comment by slandelle** [comment]
First, before going into reviewing the implementation (I'll leave this for @oleksiys as this is on the Grizzly side), there's a very important step! You didn't describe the purpose of your proposed change!!! What is this for? Is it a bug? A new feature?

File: src/main/java/com/ning/http/client/providers/grizzly/ConnectionManager.java (hunk 1 above)
Comment (elrodro83) [inline src/main/java/com/ning/http/client/providers/grizzly/ConnectionManager.java]: Yes, thanks for pointing out. Fixed it.

**Comment by elrodro83** [comment]
@slandelle Thanks, i just edited the first comment in the PR.

**Comment by oleksiys** [comment]
@slandelle can we ignore that integration failure? not sure if it's related to the change at all.
Thanks.

**Comment by slandelle** [comment]
@oleksiys Yes, you can ignore Travis on 1.9. Travis is enabled globally on the repo but there's no yaml file in 1.9 as tests would fail anyway (some old tests are localized and only work outside NA, where Travis hosts are of course based).

**Comment by oleksiys** [comment]
thank you!

**Comment by elrodro83** [comment]
Excellent. Would it be possible to get a version out with this change?
Thanks in advance.

**Comment by slandelle** [comment]
I juste release 1.9.37

**Comment by oleksiys** [comment]
thank you!

**Comment by elrodro83** [comment]
Thanks to both of you.

**Pull request merged**